core-bundle/src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -1112,10 +1113,12 @@ public static function trimsplit($strPattern, $strString)
 	 */
 	public static function stripRootDir($path)
 	{
-		$projectDir = System::getContainer()->getParameter('kernel.project_dir');
+		// Compare normalized version of the paths
+		$projectDir = Path::normalize(System::getContainer()->getParameter('kernel.project_dir'));
+		$normalizedPath = Path::normalize($path);
 		$length = \strlen($projectDir);
 
-		if (strncmp($path, $projectDir, $length) !== 0 || \strlen($path) <= $length || ($path[$length] !== '/' && $path[$length] !== '\\'))
+		if (strncmp($normalizedPath, $projectDir, $length) !== 0 || \strlen($normalizedPath) <= $length || $normalizedPath[$length] !== '/')
```

Comment:
This change seems unrelated. Am I missing something?

Also – while we are at it – why are you not using `Path::isBasePath()`?

Reply:
It's needed because due to the normalization `\` &rarr; `/` together with legacy code paths using `\`, stripping the root dir could break on windows (analogous to #1902, https://github.com/contao/contao/pull/1902/commits/48b1f8998aca6a23c268927d6bf4666893421cb6). See this CI run: https://github.com/contao/contao/pull/2516/checks?check_run_id=2088691955

> Also – while we are at it – why are you not using Path::isBasePath()?

There are some edge cases to this. We fixed the same issue in https://github.com/contao/contao/pull/1918. Now the implementation should be identical to the upstream versions.